hazelcast/include/hazelcast/client/serialization/pimpl/compact/compact_impl.h

```diff
@@ -596,18 +652,72 @@ T inline compact_stream_serializer::read(object_data_input& in)
         };
         BOOST_THROW_EXCEPTION(exception);
     }
+
     compact_reader reader = create_compact_reader(*this, in, schema);
     return hz_serializer<T>::read(reader);
 }
 
+generic_record::generic_record compact_stream_serializer::read_generic_record(object_data_input& in)
```

Comment:
This function might be inline too. By the way, these two methods are placed between compact_stream_serializer::read and compact_stream_serializer::write. For easy tracing, maybe it woould be better to move these functions after compact_stream_serializer::write